neo/Consensus/ConsensusContext.cs

```diff
@@ -231,15 +259,7 @@ public ConsensusPayload MakeRecoveryMessage()
             PrepareRequest prepareRequestMessage = null;
             if (TransactionHashes != null)
             {
-                prepareRequestMessage = new PrepareRequest
-                {
-                    ViewNumber = ViewNumber,
-                    TransactionHashes = TransactionHashes,
-                    Nonce = Nonce,
-                    NextConsensus = NextConsensus,
-                    MinerTransaction = (MinerTransaction)Transactions[TransactionHashes[0]],
-                    Timestamp = Timestamp
-                };
+                prepareRequestMessage = PreparationPayloads[PrimaryIndex].GetDeserializedMessage<PrepareRequest>();
```

Comment:
We should check if `PreparationPayloads[PrimaryIndex]` it's null

Reply:
Otherwise we should do the previous code

Reply:
We have this check 
https://github.com/neo-project/neo/blob/add411b88487da1eb0ea0f49d55be009487a4a22/neo/Consensus/ConsensusContext.cs#L260
if `TransactionHashes` is not `null`, it means node has received `PrepareRequest`.